http_parser.cc

```diff
@@ -0,0 +1,104 @@
+#include "http_parser.h"
+#include <string>
+#include <vector>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+HttpParser* HttpParser::MakeHttpParser(const char* const raw_req)
```

Comment:
I'm pretty sure async_read_some when it has the max size takes care of this for us.